boot/seal.go

```diff
@@ -48,23 +48,23 @@ var (
 )
 
 // Hook functions setup by devicestate to support device-specific full
-// disk encryption implementations.
+// disk encryption implementations. The state much be locked when these
+// functions are called.
 var (
 	HasFDESetupHook = func() (bool, error) {
 		return false, nil
 	}
-	RunFDESetupHook = func(op string, params *FdeSetupHookParams) error {
-		return fmt.Errorf("internal error: RunFDESetupHook not set yet")
+	RunFDESetupHook = func(op string, params *FDESetupHookParams) ([]byte, error) {
+		return nil, fmt.Errorf("internal error: RunFDESetupHook not set yet")
 	}
 )
 
-// FdeSetupHookParams contains the inputs for the fde-setup hook
-type FdeSetupHookParams struct {
+// FDESetupHookParams contains the inputs for the fde-setup hook
+type FDESetupHookParams struct {
```

Comment:
should this be exported?

Reply:
same question

Reply:
Ups, sorry. I meant to reply to this earlier. Right now this is part of what we export to devicestate:
```
overlord/devicestate/devicemgr.go:func (m *DeviceManager) runFDESetupHook(op string, params *boot.FDESetupHookParams) ([]byte, error) {
```
or am I misreading the question?

Reply:
ah, silly me, sorry